src/map/skill.c

```diff
@@ -3338,7 +3338,10 @@ static int skill_attack(int attack_type, struct block_list *src, struct block_li
 			case SR_KNUCKLEARROW:
 				if( skill->blown(dsrc,bl,dmg.blewcount,dir,0) && !(flag&4) ) {
 					short dir_x, dir_y;
-					Assert_ret(dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX);
+					if (!(dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX)) {
```

Comment:
Consider using `Assert_chk` here instead of a double assertion.

Reply:
Will do so. Done in a local commit.